### PR TITLE
Check Validity of .unfinished_awake_periods

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Create test to demonstrate that the scope  
  .unfinished_awake_periods is actually  
  scoping to the correct content.
* After some consternation, an arel solution,  
  while outer joins are possible, was  
  not found and/or written.

[Finishes: #88451248]

### CODE REVIEW

File: spec/models/awake_period_spec.rb [hunk 1]
```diff
@@ -59,5 +59,23 @@
 
       expect(scoped_awake_periods.count).to eq 2
     end
+
+    it ".unfinished_awake_periods returns awake periods for which there are no activities with corresponding start_time" do
+      unfinished_awake_periods_count = participant3.unfinished_awake_periods.count
```
Comment (ericcf) [inline spec/models/awake_period_spec.rb]: @michaelwehrley what do you think about using the `change` matcher? https://www.relishapp.com/rspec/rspec-expectations/v/3-2/docs/built-in-matchers/change-matcher

File: spec/models/awake_period_spec.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require "rails_helper"
 
 describe AwakePeriod do
-  fixtures :participants, :awake_periods
+  fixtures :participants, :awake_periods, :activity_types
 
   describe "validations" do
```
Comment (ericcf) [inline spec/models/awake_period_spec.rb]: @michaelwehrley a more conventional naming scheme here would be
```
describe "#unfinished_awake_periods" do
...
```
and then
```
it "returns awake periods for which there are no activities with corresponding start_time"
```
`#` denotes an instance method and `.` denotes a class method.

File: spec/models/awake_period_spec.rb (hunk 1 above)
Comment (michaelwehrley) [inline spec/models/awake_period_spec.rb]: sounds good to me.

File: spec/models/awake_period_spec.rb (hunk 2 above)
Comment (michaelwehrley) [inline spec/models/awake_period_spec.rb]: ahh yes, this one got me b/c it is a scope on an instance.  Good catch.

**Comment by michaelwehrley** [comment]
@ericcf Thanks! :ice_cream:

**Pull request merged**